packages/components/organisms/f-checkout/src/mixins/api.mixin.js

```diff
@@ -0,0 +1,86 @@
+import { mapActions } from 'vuex';
+import { VUEX_CHECKOUT_MODULE } from '../constants';
+import apiController from '../services/apiController';
+
+export default {
+    data () {
+        return {
+            registrationSource: 'Guest'
+        }
+    },
+
+    computed: {
+        firstName () {
+            return this.customer.firstName;
+        },
+
+        lastName () {
+            return this.customer.lastName;
+        },
+
+        emailAddress () {
+            return this.customer.email;
+        },
+
+        language() {
+            return this.$i18n.locale;
+        },
+
+        currentPostcode () {
+            return this.$cookies.get('je-location')
+        }
+    },
+
+    methods: {
+        ...mapActions(VUEX_CHECKOUT_MODULE, [
+            'createGuestUser',
+            'getBasket',
+            'getCheckout',
+            'getAvailableFulfilment',
+            'getAddress',
+            'getCustomer',
+        ]),
+
+        createConfig (endpoint) {
+            const config  = apiController[endpoint].config;
+            let additionalConfig = {};
+
+            config?.forEach(item => {
+                additionalConfig[item] = this[item]
+            });
+
+            const baseConfig = {
+                url: this[`${endpoint}Url`],
+                timeout: this.checkoutTimeout
+            };
+
+            return additionalConfig ? {...baseConfig, ...additionalConfig} : baseConfig;
```

Comment:
additionalConfig will always be truthy here. As it's default state is an empty object, you can just

```suggestion
            return { ...baseConfig, ...additionalConfig };
```